csharp/src/Apache.Arrow/RecordBatch.Builder.cs

```diff
@@ -0,0 +1,149 @@
+
+            private TArray Build<TArray, TArrayBuilder>(Action<TArrayBuilder> action)
+                where TArray: IArrowArray
+                where TArrayBuilder: IArrowArrayBuilder<TArray>, new()
+            {
+                if (action == null)
+                {
+                    return default;
+                }
+
+                var builder = new TArrayBuilder();
+                action(builder);
+
+                return builder.Build(_memoryPool);
+            }
+        }
+
+        public class Builder
+        {
+            private readonly MemoryPool _memoryPool;
+            private readonly Schema.Builder _schema;
+            private readonly LinkedList<IArrowArray> _arrays;
+
+            public Builder(MemoryPool memoryPool = default)
+            {
+                _memoryPool = memoryPool ?? MemoryPool.Default.Value;
+                _schema = new Schema.Builder();
+                _arrays = new LinkedList<IArrowArray>();
+            }
+
+            public RecordBatch Build()
+            {
+                var schema = _schema.Build();
+                var length = _arrays.Max(x => x.Length);
+                var batch = new RecordBatch(schema, _arrays, length);
+
+                return batch;
+            }
+
+            public Builder Clear()
+            {
+                _schema.Clear();
+                _arrays.Clear();
+                return this;
+            }
+
+            public Builder Append(RecordBatch batch)
+            {
+                foreach (var field in batch.Schema.Fields)
+                {
+                    _schema.Field(field.Value);
+                }
+
+                foreach (var array in batch.Arrays)
+                {
+                    _arrays.AddLast(array);
+                }
+
+                return this;
+            }
+
+            public Builder Append<TArray>(string name, bool nullable, IArrowArrayBuilder<TArray> builder)
+                where TArray: IArrowArray
+            {
+                return builder == null 
+                    ? this 
+                    : Append(name, nullable, builder.Build(_memoryPool));
+            }
+
+            public Builder Append<TArray>(string name, bool nullable, TArray array)
+                where TArray: IArrowArray
+            {
+                if (string.IsNullOrWhiteSpace(name)) throw new ArgumentNullException(nameof(name));
+                if (array == null) return this;
+
+                _arrays.AddLast(array);
+
+                _schema.Field(f => f
+                    .Name(name)
+                    .Nullable(nullable)
+                    .DataType(array.Data.DataType));
+
+                return this;
+            }
+
+            public Builder Append<TArray>(string name, bool nullable, Func<ArrayBuilder, TArray> action)
+                where TArray: IArrowArray
+            {
+                if (action == null) return this;
+
+                var arrayBuilder = new ArrayBuilder(_memoryPool);
```

Comment:
Do we need to create a new instance of `ArrayBuilder` for every call to this method?